modules/core/src/minmax.cpp

```diff
@@ -71,33 +71,815 @@ minMaxIdx_( const T* src, const uchar* mask, WT* _minVal, WT* _maxVal,
     *_maxVal = maxVal;
 }
 
+#if CV_SIMD128
+#ifdef _MSC_VER
+    #define forceinline __forceinline
+#elif defined(__GNUC__)
+    #define forceinline inline __attribute__((__always_inline__))
+#elif defined(__CLANG__)
+    #if __has_attribute(__always_inline__)
+        #define forceinline inline __attribute__((__always_inline__))
+    #else
+        #define forceinline inline
+    #endif
+#else
+    #define forceinline inline
+#endif
+
+template<typename T, typename WT> forceinline void
+minMaxIdx_init( const T* src, const uchar* mask, WT* minval, WT* maxval,
+                size_t* minidx, size_t* maxidx, WT &minVal, WT &maxVal,
+                size_t &minIdx, size_t &maxIdx, const WT minInit, const WT maxInit,
+                const int nlanes, int len, size_t startidx, int &j, int &len0 )
+{
+    len0 = len & -nlanes;
+    j = 0;
+
+    minVal = *minval, maxVal = *maxval;
+    minIdx = *minidx, maxIdx = *maxidx;
+
+    // To handle start values out of range
+    if ( maxVal < minInit || minVal > maxInit )
```

Comment:
It looks like `minInit` and `maxInit` are always set to type boundaries, while `minVal` and `maxVal` are corresponding type variables. I suppose this condition never could be satisfied.

Reply:
Actually minVal/maxVal are commonly set to INT_MIN/INT_MAX and are 'int' types.  This means they are out of range for 'char' and 'short' types until they initialized.  The same is true for float/double - they are commonly initialized to +/- infinity.

So this condition is needed.

Reply:
Sorry, missed that. However, e.g. for u8 there could be other case that won't be handled right:
`minVal == -1; maxVal == 256;`

Reply:
I'll add 'minVal < minInit || maxVal > maxInit)